src/Engine/Graphics/BspRenderer.cpp

```diff
@@ -15,25 +15,57 @@ void BspRenderer::AddFace(const int node_id, const int uFaceID) {
     assert(uFaceID > -1 && uFaceID < pIndoor->pFaces.size() && "please report with a nearby save file");
 
     BLVFace *pFace = &pIndoor->pFaces[uFaceID];
+    pFace->uAttributes |= FACE_SeenByParty;
+
+    // NOTE(yoctozepto): the below happens, e.g., on various stairs
+    // TODO(yoctozepto): might be nice to check if the vertices actually form a plane and not a line;
+    //                   this could be done when loading the location and filtering out such broken faces
+    if (pFace->uNumVertices < 3) {
+        return;  // nothing to render
+    }
+
+    const BspRenderer_ViewportNode *currentNode = &nodes[node_id];
+
+    // check if any triangle of the face can be seen
+
+    static RenderVertexSoft originalFaceVertices[64];
+    static RenderVertexSoft clippedFaceVertices[64];
+
+    // TODO(yoctozepto): are face vertices consecutive? are face vertices shared/overlapping?
```

Comment:
Usually, not always. The polygon geometry is stored as triangle fans.

Reply:
Thanks, will have this in mind when playing around further.